api/specs.proto

```diff
@@ -258,6 +258,12 @@ message ContainerSpec {
 	// task will exit and a new task will be rescheduled elsewhere. A container
 	// is considered unhealthy after `Retries` number of consecutive failures.
 	HealthConfig healthcheck = 16;
+
+	message SecurityConfig {
+	    string userns = 30;
+	    string credential_spec = 31 [(gogoproto.customname) = "CredentialSpec"];
```

Comment:
`registry_auth` was a bit of an exception ...

We added it knowing in advance that we'd deprecate it (in favor of using secrets to store registry auth) soooo we picked a high number we'd be unlikely to need in the future. Sorry you had to witness that :) We were trying to hide it under the rug.